catroid/src/main/java/org/catrobat/catroid/content/actions/ComeToFrontAction.java

```diff
@@ -38,15 +37,7 @@
 	protected void update(float delta) {
 
 		List<Sprite> spriteList = ProjectManager.getInstance().getCurrentProject().getSpriteListWithClones();
-		int actualSpriteZIndex = sprite.look.getZIndex();
-
-		for (int i = 0; i < spriteList.size(); i++) {
-			if (spriteList.get(i).look.getZIndex() > actualSpriteZIndex) {
-				spriteList.get(i).look.setZIndex(spriteList.get(i).look.getZIndex()
-						+ Constants.Z_INDEX_NUMBER_VIRTUAL_LAYERS - 1);
-			}
-		}
-		sprite.look.setZIndex(spriteList.size() + Constants.Z_INDEX_NUMBER_VIRTUAL_LAYERS - 1);
+		sprite.look.setZIndex(spriteList.size());
```

Comment:
I guess we can't remove `…+ Constants.Z_INDEX_NUMBER_VIRTUAL_LAYERS - 1` here. Even though `Z_INDEX_NUMBER_VIRTUAL_LAYERS` is currently `1`, it may change in the future.

This case is different from `spriteList.get(i).look.setZIndex(spriteList.get(i).look.getZIndex() +…` where the new z-index is calculated relatively. When you use `getZIndex`, you already have an index which considers all virtual layers.